### PR TITLE
New code analysis nupkgs

### PR DESCRIPTION
**Pull request by mavasani**
Move the roslyn-analyzers repo to latest Microsoft.CodeAnalysis NuGet packages to unblock further testing of new IOperation APIs

@dotnet/roslyn-analysis @JohnHamby @genlu

### CODE REVIEW

File: src/System.Runtime.Analyzers/Core/TestForNaNCorrectly.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public sealed class TestForNaNCorrectlyAnalyzer : DiagnosticAnalyzer
                                                                              s_localizableMessage,
                                                                              DiagnosticCategory.Usage,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: true,
+                                                                             isEnabledByDefault: false,
```
Comment (srivatsn) [inline src/System.Runtime.Analyzers/Core/TestForNaNCorrectly.cs]: Why this change?

Comment (mavasani) [inline src/System.Runtime.Analyzers/Core/TestForNaNCorrectly.cs]: Ah, thanks for catching this. This analyzer was crashing before this package upgrade and I had turned it off locally. Did not intend to include this - I'll revert.

Comment (lgolding) [inline src/System.Runtime.Analyzers/Core/TestForNaNCorrectly.cs]: Hey, yeah, that's my analyzer ;-)

Seriously though, what I actually hoped is that with this change, we could then re-enable one of the unit tests (`CSharpNoDiagnosticForComparisonWithBadExpression`), which was skipped due to dotnet/roslyn#8114. Instead, the analyzer is now disabled. What happened?

Comment (lgolding) [inline src/System.Runtime.Analyzers/Core/TestForNaNCorrectly.cs]: Awesome! @mavasani, can you re-enable that test, then, while you're at it?

**Comment by srivatsn** [comment]
:+1:

File: src/System.Runtime.Analyzers/Core/TestForNaNCorrectly.cs (hunk 1 above)
Comment (mavasani) [inline src/System.Runtime.Analyzers/Core/TestForNaNCorrectly.cs]: Done, analyzer and test have both been re-enabled.
